packages/config-loader/src/lib/schema/collect.ts

```diff
@@ -40,34 +41,47 @@ const req =
  */
 export async function collectConfigSchemas(
   packageNames: string[],
+  packagePaths: string[],
 ): Promise<ConfigSchemaPackageEntry[]> {
   const visitedPackages = new Set<string>();
   const schemas = Array<ConfigSchemaPackageEntry>();
   const tsSchemaPaths = Array<string>();
   const currentDir = await fs.realpath(process.cwd());
 
-  async function processItem({ name, parentPath }: Item) {
-    // Ensures that we only process each package once. We don't bother with
-    // loading in schemas from duplicates of different versions, as that's not
-    // supported by Backstage right now anyway. We may want to change that in
-    // the future though, if it for example becomes possible to load in two
-    // different versions of e.g. @backstage/core at once.
-    if (visitedPackages.has(name)) {
-      return;
-    }
-    visitedPackages.add(name);
+  async function processItem(item: Item) {
+    let pkgPath = item.packagePath;
 
-    let pkgPath: string;
-    try {
-      pkgPath = req.resolve(
-        `${name}/package.json`,
-        parentPath && {
-          paths: [parentPath],
-        },
-      );
-    } catch {
-      // We can somewhat safely ignore packages that don't export package.json,
-      // as they are likely not part of the Backstage ecosystem anyway.
+    if (pkgPath) {
+      const pkgExists = await fs.pathExists(pkgPath);
+      if (!pkgExists) {
+        return;
+      }
+    } else if (item.name) {
+      const { name, parentPath } = item;
+
+      // Ensures that we only process each package once. We don't bother with
+      // loading in schemas from duplicates of different versions, as that's not
+      // supported by Backstage right now anyway. We may want to change that in
+      // the future though, if it for example becomes possible to load in two
+      // different versions of e.g. @backstage/core at once.
```

Comment:
```suggestion
      // different versions of e.g. @backstage/core-app-api at once.
```

right?

Reply:
Gonna do a followup PR instead and fix what the comment is talking about, since I think we actually want to change that now

Reply:
#7350